_guides/legion/bios/bios-recovery.md

```diff
@@ -1,32 +1,66 @@
 ---
 layout: wiki
-title: Bios Recovery
-description: "Recovering bricked BIOS in Legions without a backup ROM"
+title: BIOS Recovery
+description: "Recovering bricked BIOS in Legions using a SPI"
 parent: Bios
 grand_parent: Legion
-contributors:
-discord: ['LABØ!#5022']
+contributors: ['labodj'] 
 ---
 
-# Recovering bricked BIOS in Legions without a backup ROM
+# Recovering bricked BIOS in Legions using a SPI
 
 ## My Legion Laptop BIOS is screwed up and I don't have a BIOS backup, how can I try to restore it?
 
-Basically a 3.3V SPI programmer like the CH341A is needed and also knowledge about it is required. This [video](https://www.youtube.com/watch?v=4qX2zihB6UE){:target="_blank"} should help to understand how the programmer works, if you still cant figure out how this programmer works, then this guide isn't for you. 
+A 3.3V SPI programmer like the CH341A is needed and also knowledge about it is required. This [video](https://www.youtube.com/watch?v=4qX2zihB6UE){:target="_blank"} should help to understand how the programmer works, if you still can't figure out how SPI works, then this guide isn't for you.
 
-## Steps to flash the bios
+### Step 0: Prepare your tools
+
+Prepare your SPI, SPI software, install HxD and 7-Zip.
+
+### Step 1: Backup
+
+The very first thing to do is to have a backup of your BIOS, there are 2 paths here:
+
+- You already have a backup at least of BIOS region (usually 16MB), go to step 2, and skip step 3.
+- You don't have a backup, continue here...
+
+If you don't have a backup (very bad thing!) you should make one now, it's not mandatory but it's required if you want to restore your serials/OS keys afterwards.
+
+Obviously if your laptop is bricked to make a new backup you'll need to perform a SPI DUMP from the IC containing BIOS region (usually 16MB), dump it and ***verify the dump against IC content*** with the software you're using (NeoProgrammer 2.2.0.8 was used during my testing).
+
+### Step 2: Clean vendor BIOS file
 
 You can build a config-less, serial-less BIOS using the one provided by Lenovo following these steps
 
-- First of all, it's better to dump the current non-working BIOS using the SPI programmer. **DUMP IT** and **VERIFY** the dump with the software you're using (NeoProgrammer 2.2.0.8 was used in my testing). You can recover your own serials/os keys from the broken BIOS, there will be another guide for that, but you need to have a good dump for that.
-- Install HxD, 7-Zip
 - Download the BIOS from Lenovo site
 - Start the downloaded ``.exe`` and select to extract it in a convenient folder (eg: desktop/bios/)
```

Comment:
I think newer bios versions will need to use innoextract as it checks compatibility before it self extracts.

Reply:
Thanks for the suggestion, "protected" setups procedure added in b0300465b1dab7bfcb693be47ce5f725519a9080